sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileFormat.scala

```diff
@@ -276,10 +275,16 @@ object FileFormat {
           // the modificationTime from the file is in millisecond,
           // while internally, the TimestampType `file_modification_time` is stored in microsecond
           row.update(i, fileModificationTime * 1000L)
-        case ROW_INDEX =>
-          // Do nothing. Only the metadata fields that have identical values for each row of the
-          // file are set by this function, while fields that have different values (such as row
-          // index) are set separately.
+        case other =>
+          // Other metadata columns use the file-provided value (if any). As a courtesy, convert any
+          // normal strings to the required [[UTF8String]].
+          //
+          // TODO(frj): Do we need to potentially support value-producing functions?
```

Comment:
Could we file a JIRA and fix it like `TODO(SPARK-XXXXX)` instead?

Reply:
Removing for now, because the TODO doesn't add any meaningful information. 
If/when the need arises, it will be obvious enough that this code needs to change.